staging/src/k8s.io/legacy-cloud-providers/azure/azure_client.go

```diff
@@ -191,7 +191,10 @@ func newAzVirtualMachinesClient(config *azClientConfig) *azVirtualMachinesClient
 
 func (az *azVirtualMachinesClient) CreateOrUpdate(ctx context.Context, resourceGroupName string, VMName string, parameters compute.VirtualMachine, source string) (resp *http.Response, err error) {
 	// /* Write rate limiting */
+	mc := newMetricContext("vm", "create_or_update", resourceGroupName, az.client.SubscriptionID, source)
+	mc.Count()
```

Comment:
The count here is actually not true. When `mc.Count()` is invoked and ratelimited by the following line, then the requests won't be sent to Azure. 

I think we should move this before `az.client.CreateOrUpdate()`. Same all other clients.

Reply:
There are two metrics - one which is the count of operations  which is actually made by various components. And then the count which are ratelimited. We want to be able to know how many calls we are making and how many we are rate limiting. Just knowing how many were sent to azure is not enough information to arrive at conclusions. This gives us a better picture. Also, we can  gather the information of how many went to azure from these two counters.

Reply:
I mean we should expose the counts sent to Azure as well as the ratelimited. The total count could be calculated by sum both.